pkg/controllers/tagging/tagging_controller.go

```diff
@@ -31,19 +31,25 @@ import (
 
 // workItem contains the node and an action for that node
 type workItem struct {
-	node   *v1.Node
-	action func(node *v1.Node) error
+	node           *v1.Node
+	action         func(node *v1.Node) error
+	requeuingCount int
 }
 
+const (
+	maxRequeuingCount = 10
```

Comment:
Do we know what this translates into in terms of retry duration?

Reply:
We are using the [DefaultControllerRateLimiter](https://github.com/kubernetes/client-go/blob/4fc3881b7ffe4a5c1c7c3cffc7dd9d9667989730/util/workqueue/default_rate_limiters.go#L39), which use the worst case of the 2 limiters:
- [ItemExponentialFailureRateLimiter](https://github.com/kubernetes/client-go/blob/4fc3881b7ffe4a5c1c7c3cffc7dd9d9667989730/util/workqueue/default_rate_limiters.go#L67), with a base delay of 5 milliseconds and max delay of 1000 seconds
- [BucketRateLimiter](https://github.com/kubernetes/client-go/blob/4fc3881b7ffe4a5c1c7c3cffc7dd9d9667989730/util/workqueue/default_rate_limiters.go#L48), with a limit of 10, and 100 burst.
If not counting BucketRateLimiter cause that's a different beast, the work items would have to wait anywhere from 50 milliseconds to 10000 seconds in retry time (excluding work time and queue wait time as they can differ). However, the real avg number is a bit more complicated depending on the number of items in the queue, if the item has been enqueue many times, which rate limiter was used. So, `When` to add an item back in the queue depends on either [BucketRateLimiter](https://github.com/kubernetes/client-go/blob/4fc3881b7ffe4a5c1c7c3cffc7dd9d9667989730/util/workqueue/default_rate_limiters.go#L54) or [ItemExponentialFailureRateLimiter](https://github.com/kubernetes/client-go/blob/4fc3881b7ffe4a5c1c7c3cffc7dd9d9667989730/util/workqueue/default_rate_limiters.go#L89) is selected as worst case. However, I wouldn't expect the worst case of 10000 to ever happens, though we need metrics data to fine tune this as it goes I assume.